src/app/components/music-controls/music-controls.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component } from '@angular/core';
+import { AudioService } from '../../services/audio.service';
+import { StreamState, ITrack } from '../../types/interfaces';
+import { displayMillisecInMinSec } from '../../utils/utils';
+
+@Component({
+  selector: 'app-music-controls',
+  templateUrl: './music-controls.component.html',
+  styleUrls: ['./music-controls.component.scss']
+})
+export class MusicControlsComponent {
+  public state: StreamState;
+  public currentAudioID: string;
+  public currentTrackList: ITrack[];
+  public currentTrack: ITrack;
+  public isAutorenewed: boolean;
+  public isShuffled: boolean;
+  public isTracksListEnd: boolean = false;
+  public isVolumeOff: boolean = false;
+
+  constructor(
+    private audioService: AudioService
+  ) {
+    this.listenAudioService();
+    this.listenAudioChanges();
+    this.listenTrackListChanges();
+    this.isAutorenewed = this.audioService.getAutorenew();
+    this.isShuffled = this.audioService.getShuffle();
```

Comment:
try to avoid using constructor to initialize internal methods - use lifecycle hook `ngOnInit` instead.